### ISSUE TITLE
Strapi Startup unexpectedly wants to alter tables

### ISSUE THREAD

**Issue by seth146** [strapi/strapi#13401]
### Required System information

- Node.js version: 16.14.2
- NPM version: 8.5.0
- Strapi version: 4.1.10
- Database: PostgreSQL 14.1 on aarch64-unknown-linux-gnu, compiled by gcc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0, 64-bit ( provisioned via Supabase)
- Operating system: macOS Monterey 12.2.1 (locally)

### Describe the bug

I use a Supabase-provisioned Postgres DB as a backing DB for the Strapi instance as well as the main backend DB for my app. In the DB, I link my backend tables to the Strapi tables via foreign keys to have easy access to the content provided by Strapi.

When starting the server via `npm run develop`, I run into the following error on the foreign keys:

```
[2022-05-27 10:26:51.748] debug: ⛔️ Server wasn't able to start properly.
[2022-05-27 10:26:51.754] error: alter table "groups" drop constraint "groups_pathway_id_fkey" - must be owner of table groups
error: alter table "groups" drop constraint "groups_pathway_id_fkey" - must be owner of table groups
    at Parser.parseErrorMessage (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:526:28)
    at Socket.emit (node:domain:475:12)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```

Table `groups` is an app-specific table, which is linked to a Strapi-owned table `pathways` via a foreign key.

*EDIT*: This even happens with foreign keys between tables that are not Strapi related at all - table `users_groups_links` is my custom backend table, and the foreign key relations are linking it with other custom backend tables. Yet, the same error is thrown:

```
[2022-05-27 11:05:11.217] debug: ⛔️ Server wasn't able to start properly.
[2022-05-27 11:05:11.219] error: alter table "users_groups_links" drop constraint "users_groups_links_group_id_fkey" - must be owner of table users_groups_links
error: alter table "users_groups_links" drop constraint "users_groups_links_group_id_fkey" - must be owner of table users_groups_links
    at Parser.parseErrorMessage (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/Users/pavol/Documents/ksebe/group-app/ally-app-cms/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:526:28)
    at Socket.emit (node:domain:475:12)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```

### Steps to reproduce the behavior

1. Have foreign key links between Strapi tables and other backend tables
2. Try to start the server using `npm run develop`
3. Fail with the error above

### Expected behavior

I expect that Strapi would not try to alter tables on my behalf at the startup - even more for tables that are not Strapi-related at all - and that it would respect the foreign key relations.

### Additional context

Creating a new issue in lieu of [11421](https://github.com/strapi/strapi/issues/11421), as requested by @derrickmehaffy.

**Comment by derrickmehaffy** [comment]
> I use a Supabase-provisioned Postgres DB as a backing DB for the Strapi instance as well 

We do not support running Strapi on a previously created database from a different application and especially not running two applications in parallel on the same database.

I'll give you some time to respond on this but most likely I'll need to close this issue as won't fix.
